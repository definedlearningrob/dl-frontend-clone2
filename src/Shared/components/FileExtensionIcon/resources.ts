export const imageExtensions = ['jpg', 'jpeg', 'png', 'gif'];
export const videoExtensions = ['wma', 'mpg', 'flv', 'avi', 'mp4', 'webm'];
export const documentExtensions = ['pdf', 'doc', 'docx', 'txt', 'csv', 'rtf', 'html'];
export const presentationExtensions = ['ppt', 'pptx'];
export const sheetExtensions = ['xls', 'xlsx'];
