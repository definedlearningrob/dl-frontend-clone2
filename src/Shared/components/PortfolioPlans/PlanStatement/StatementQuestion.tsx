import { isEmpty } from 'lodash-es';

import { useAnswerPlanGroupStatementQuestion } from '@shared/graphql/shared/hooks/useAnswerPlanGroupStatementQuestion';

import { PlanStatementQuestion } from '../PlanStatementQuestion/PlanStatementQuestion';
import { TPlanStatementQuestion } from '../types';

type Props = {
  question: TPlanStatementQuestion | null;
  evaluationId: string;
};

export const StatementQuestion = ({ question, evaluationId }: Props) => {
  const [answerStatementQuestion] = useAnswerPlanGroupStatementQuestion();

  if (!question || isEmpty(question.text)) {
    return null;
  }

  const { text, answer, options, questionType, id } = question;

  const handleSubmit = async (answer: string[]) => {
    await answerStatementQuestion({ evaluationId, questionId: id, answer });
  };

  return (
    <PlanStatementQuestion
      answer={answer?.answer}
      options={options}
      questionType={questionType}
      text={text}
      onSubmit={handleSubmit}
    />
  );
};
