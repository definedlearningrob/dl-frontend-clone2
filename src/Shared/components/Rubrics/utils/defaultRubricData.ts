export const defaultRubricData = {
  headings: [
    { id: 'heading-1', name: 'rubric.careerPathway.name', multiplier: 1 },
    { id: 'heading-2', name: 'rubric.projectGoal.name', multiplier: 1 },
    { id: 'heading-3', name: 'rubric.projectRole.name', multiplier: 1 },
    { id: 'heading-4', name: 'rubric.jobProductQuality.name', multiplier: 1 },
  ],
  criteriaLabels: [
    { displayName: 'test label', id: 'criteria-label-1', score: 1 },
    { displayName: null, id: 'criteria-label-2', score: 2 },
    { displayName: null, id: 'criteria-label-3', score: 3 },
    { displayName: null, id: 'criteria-label-4', score: 4 },
  ],
  criterias: [
    {
      id: 'criteria-1-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.1',
    },
    {
      id: 'criteria-1-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.2',
    },
    {
      id: 'criteria-1-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.3',
    },
    {
      id: 'criteria-1-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.4',
    },
    {
      id: 'criteria-2-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.1',
    },
    {
      id: 'criteria-2-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.2',
    },
    {
      id: 'criteria-2-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.3',
    },
    {
      id: 'criteria-2-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.4',
    },
    {
      id: 'criteria-3-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.1',
    },
    {
      id: 'criteria-3-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.2',
    },
    {
      id: 'criteria-3-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.3',
    },
    {
      id: 'criteria-3-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.4',
    },
    {
      id: 'criteria-4-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.1',
    },
    {
      id: 'criteria-4-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.2',
    },
    {
      id: 'criteria-4-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.3',
    },
    {
      id: 'criteria-4-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.4',
    },
  ],
};
