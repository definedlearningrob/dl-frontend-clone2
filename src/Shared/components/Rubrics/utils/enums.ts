export enum RUBRIC_TYPE {
  VIEWER = 'VIEWER',
  GRADER = 'GRADER',
  PREVIEW = 'PREVIEW',
}
