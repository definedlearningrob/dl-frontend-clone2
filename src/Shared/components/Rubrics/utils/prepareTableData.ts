import { v4 as uuid } from 'uuid';

import {
  FillMissingCriteriasArguments,
  RubricHeadingIdPairs,
  RubricCriteria,
  RubricData,
} from './types';

const fillMissingCriteria = ({
  headings,
  criteriaLabels,
  criterias,
}: FillMissingCriteriasArguments): RubricCriteria[] => {
  const labelIds = criteriaLabels.map((label) => label.id);
  const headingIds = headings.map((heading) => heading.id);

  const labelHeadingIdPairs: RubricHeadingIdPairs = labelIds.reduce(
    (acc: RubricHeadingIdPairs, criteriaLabelId) => {
      const pairsForFirst = headingIds.map((headingId) => ({ headingId, criteriaLabelId }));
      acc.push(...pairsForFirst);

      return acc;
    },
    []
  );

  return labelHeadingIdPairs.map((pair) => {
    const foundCriteria = criterias.find(
      ({ rubricCriteriaLabelId, rubricHeadingId }) =>
        rubricCriteriaLabelId === pair.criteriaLabelId && rubricHeadingId === pair.headingId
    );

    return (
      foundCriteria || {
        uuid: uuid(),
        text: '',
        rubricCriteriaLabelId: pair.criteriaLabelId,
        rubricHeadingId: pair.headingId,
      }
    );
  });
};

export const prepareTableData = (rubric: RubricData): RubricData => {
  const { criterias, headings, criteriaLabels } = rubric;

  return {
    ...rubric,
    criterias: fillMissingCriteria({ headings, criteriaLabels, criterias }),
  };
};
