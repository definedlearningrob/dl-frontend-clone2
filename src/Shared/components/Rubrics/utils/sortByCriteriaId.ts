import { RubricCriteria, RubricCriteriaLabel } from './types';

export const byCriteriaLabelIds =
  (criteriaLabels: RubricCriteriaLabel[]) =>
  (criteriaA: RubricCriteria, criteriaB: RubricCriteria) =>
    criteriaLabels.findIndex((label) => label.id === criteriaA.rubricCriteriaLabelId) -
    criteriaLabels.findIndex((label) => label.id === criteriaB.rubricCriteriaLabelId);
