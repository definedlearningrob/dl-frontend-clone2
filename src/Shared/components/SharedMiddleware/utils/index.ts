export * from './parseSharedResource';
export * from './getResourceLink';
