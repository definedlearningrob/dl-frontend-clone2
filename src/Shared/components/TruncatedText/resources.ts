export const OFFSET_HEIGHT = 63;
