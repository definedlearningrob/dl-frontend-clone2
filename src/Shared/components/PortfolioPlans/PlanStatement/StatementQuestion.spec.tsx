import { screen, waitFor } from '@testing-library/dom';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import userEvent from '@testing-library/user-event';

import { renderWithI18N } from '@dc/utils/test';

import { STATEMENT_QUESTION_TYPE } from '@shared/resources/enums';
import {
  ANSWER_STATEMENT_QUESTION,
  TAnswerStatementQuestionData,
  TAnswerStatementQuestionMutationInput,
} from '@shared/graphql/shared/mutations/answerPlanGroupStatementQuestion';

import { StatementQuestion } from './StatementQuestion';

const answerStatementQuestionSpy = jest.fn();

const answerStatementQuestionMock: MockedResponse<
  TAnswerStatementQuestionData,
  TAnswerStatementQuestionMutationInput
> = {
  request: {
    query: ANSWER_STATEMENT_QUESTION,
    variables: { input: { answer: ['Example answer'], evaluationId: '1', questionId: '123' } },
  },
  result() {
    answerStatementQuestionSpy();

    return {
      data: {
        answerPlanGroupStatementQuestion: {
          answer: {
            answer: ['Example answer'],
            id: '789',
          },
        },
      },
    };
  },
};

describe('StatementQuestion', () => {
  it('initializes with correct values', () => {
    const { container } = renderWithI18N(
      <MockedProvider>
        <StatementQuestion
          evaluationId='1'
          question={{
            id: '123',
            questionType: STATEMENT_QUESTION_TYPE.SHORT_TEXT,
            options: [],
            text: 'Example question',
            answer: { answer: ['Initial answer'], id: '456' },
          }}
        />
      </MockedProvider>
    );

    const textInput = screen.getByRole('textbox');
    expect(textInput).toHaveValue('Initial answer');
    expect(textInput).toBeDisabled();

    expect(container).toMatchSnapshot();
  });

  it('does not render when text is empty string', () => {
    const { container } = renderWithI18N(
      <MockedProvider>
        <StatementQuestion
          evaluationId='1'
          question={{
            id: '123',
            questionType: STATEMENT_QUESTION_TYPE.SHORT_TEXT,
            options: [],
            text: '',
            answer: { answer: ['Initial answer'], id: '456' },
          }}
        />
      </MockedProvider>
    );

    expect(container).toBeEmptyDOMElement();
  });

  it('saves filled answer correctly', async () => {
    renderWithI18N(
      <MockedProvider mocks={[answerStatementQuestionMock]}>
        <StatementQuestion
          evaluationId='1'
          question={{
            id: '123',
            questionType: STATEMENT_QUESTION_TYPE.SHORT_TEXT,
            options: [],
            text: 'Example question',
            answer: null,
          }}
        />
      </MockedProvider>
    );

    const textInput = screen.getByRole('textbox');
    userEvent.type(textInput, 'Example answer');

    const saveButton = screen.getByRole('button', { name: 'Save' });
    userEvent.click(saveButton);

    await waitFor(() => {
      expect(answerStatementQuestionSpy).toHaveBeenCalledTimes(1);
    });
  });
});
