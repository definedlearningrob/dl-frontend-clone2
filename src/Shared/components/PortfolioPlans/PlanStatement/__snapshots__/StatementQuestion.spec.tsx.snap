// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatementQuestion initializes with correct values 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="border border-neutral-300 rounded-xs"
    >
      <div
        class="p-xs pt-x xxxl:p-sm xxxl:pb-x bg-neutral-200 rounded-t-xs text-xs xxxl:text-sm font-medium leading-lg"
      >
        Example question
      </div>
      <div
        class="flex py-sm px-x xxxl:px-sm flex-col gap-xs"
      >
        <label
          class="flex flex-col items-start text-font-primary w-full gap-xxxs"
        >
          <div
            class="w-full flex gap-xs items-center justify-between bg-white rounded-sm border border-neutral-300 hover:border-neutral-400 group focus-within:!border-primary-500 p-xs !bg-neutral-200 !border-neutral-400"
          >
            <span
              class="flex gap-xxs grow"
            >
              <input
                class="bg-transparent text-font-primary min-w-0 w-full grow w-full outline-none transition-colors autofill:shadow-[inset_0_0_0px_1000px_rgb(252,252,252)] placeholder:text-font-secondary leading-[14px] text-xxs !text-font-secondary autofill:shadow-[inset_0_0_0px_1000px_theme(colors.neutral.200)]"
                disabled=""
                name="answer"
                placeholder="Type here..."
                type="text"
                value="Initial answer"
              />
            </span>
          </div>
        </label>
        <div
          class="flex justify-end"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] w-[80px] -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
            data-testid="button"
            type="button"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Edit
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
